### ISSUE TITLE
Not list files after set "ENFORCE_BASEPATH": true

### ISSUE THREAD

**Issue by iStitch07** [danielperna84/hass-configurator#171]
Hello, what I do wrong?
I setup configurator via pip3 install

======================================================
**/etc/hass/settings.conf**
{
    "LISTENIP": "0.0.0.0",
    "PORT": 3218,
    "GIT": false,
    "BASEPATH": "/home/hass/.homeassistant/",
    "ENFORCE_BASEPATH": true,
    "SSL_CERTIFICATE": null,
    "SSL_KEY": null,
    "IGNORE_SSL": false,
    "HASS_API": "http://127.0.0.1:8123/api/",
    "HASS_WS_API": null,
    "HASS_API_PASSWORD": "Long time live access token"
    "USERNAME": null,
    "PASSWORD": null,
    "ALLOWED_NETWORKS": [],
    "ALLOWED_DOMAINS": [],
    "BANNED_IPS": [],
    "BANLIMIT": 0,
    "IGNORE_PATTERN": [],
    "DIRSFIRST": true,
    "HIDEHIDDEN": true,
    "SESAME": null,
    "SESAME_TOTP_SECRET": null,
    "VERIFY_HOSTNAME": null,
    "ENV_PREFIX": "HC_",
    "NOTIFY_SERVICE": "persistent_notification.create"
}
======================================================
**/etc/systemd/system/hass-configurator@hass.service**
[Unit]
Description=HASS-Configurator
After=network-online.target

[Service]
Type=simple
User=%i
WorkingDirectory=/home/hass/.homeassistant/
ExecStart=/usr/local/bin/hass-configurator /etc/hass/settings.conf
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=multi-user.target

======================================================

**/var/log/syslog**
Nov 10 19:41:13 axius hass-configurator[1123]: INFO:2019-11-10 19:41:13,977:hass_configurator.configurator:192.168.1.123 - "GET / HTTP/1.1" 200 -
Nov 10 19:41:14 axius hass-configurator[1123]: INFO:2019-11-10 19:41:14,854:hass_configurator.configurator:192.168.1.123 - "GET /api/listdir?path=. HTTP/1.1" 200 -
Nov 10 19:41:14 axius hass-configurator[1123]: WARNING:2019-11-10 19:41:14,855:hass_configurator.configurator:Access denied.

======================================================
<img width="341" alt="2019-11-10 в 19 46 59" src="https://user-images.githubusercontent.com/56363568/68547455-e78d7d80-03f2-11ea-96c7-38070e5bd195.png">

======================================================

If I change "ENFORCE_BASEPATH": to false all start working good, but possible to change directory and list file system, what I want to avoid.

**Comment by danielperna84** [comment]
Please try to start the configurator without the enforced basepath, then open a file within the basepath. That way the configurator will re-open the same file and path at the next startup. Then stop the configurator, enable enforce basepath, then start again. It should then start with the file you had open last and display the content of the directory.

I assume you get the error because prior to shutting down the configurator you had a file open that was outside of the basepath. The configurator remembers this and tries to access the same file when you restart the configurator, even though with the enforced basepath access to that file is now denied.

Does this solution work?

**Comment by iStitch07** [comment]
Nope. i migrated to docker and there configurator work perfect. But I special try to reproduce situation in dev environment and receive same result
Now configurator endorsed to 
"BASEPATH": "/home/projects/",
"ENFORCE_BASEPATH": true,

And any way
<img width="725" alt="Снимок экрана 2019-11-25 в 22 33 34" src="https://user-images.githubusercontent.com/56363568/69571863-046da780-0fd4-11ea-8b8d-fa5872ddf4a2.png">

Because its not affecting me now, you can close this issue without resolution, just for your information.

**Comment by victor-falcon** [comment]
I have the same problem right now. I can't navigate... :disappointed:

**Comment by victor-falcon** [comment]
There´re any ways to fully restart or reset the plugging? I uninstall and install but the problem persist.

**Comment by danielperna84** [comment]
I don't know why this is happening. As a workaround you should be able to open a specific folder manually. To do this, open the JavaScript console of your browser and execute the function `listdir('/your/path')`.

**Comment by victor-falcon** [comment]
![image](https://user-images.githubusercontent.com/238766/70472245-1456b280-1acf-11ea-8574-b0fae23b4187.png)

listdir command doesn't work.

I try to open configurator with another browser and the problem persist

**Comment by victor-falcon** [comment]
I also saw this, maybe it helps
![image](https://user-images.githubusercontent.com/238766/70472759-f3db2800-1acf-11ea-8020-161ede75f9af.png)

**Comment by danielperna84** [comment]
There are two options:
1. There seems to be a problem with the ingress-functionality of hassio. It somehow messes with the external dependencies the configurator relies on. If this is really the case, please open an issue at the [hassio-addons](https://github.com/home-assistant/hassio-addons) repository.
2. You have something like PiHole or some other content blocking mechanism active that prevents the configurator to load these external resources.

**Comment by victor-falcon** [comment]
There're another problem. Now I can load base path with `listdir('/')` but when I try to open config folder I get no response.

![image](https://user-images.githubusercontent.com/238766/71191826-78c00180-2287-11ea-93ab-025c1cdaadb5.png)
![image](https://user-images.githubusercontent.com/238766/71191782-62b24100-2287-11ea-83a4-e6dd15448e03.png)

**Comment by danielperna84** [comment]
I don't use hassio and the ingress-functionality, so I don't know what the problem could be. Please head over to the forum at https://community.home-assistant.io to see if someone there knows why this is happening.

**Comment by victor-falcon** [comment]
The problem it's not with ingress. When I access from local network with local IP it still happening.

**Comment by danielperna84** [comment]
Could you make another screenshot like above when accessing locally? And instead of the preview-tab please open the response-tab.

**Comment by ravikwow** [comment]
I start configurator.py with settings.conf file. If I set ENFORCE_BASEPATH to false then work fine and I can view all file system. But if i set ENFORCE_BASEPATH to false then I view situation on screenshot: I can not view basepath directory, but configurator.py open file from previous session from basepath directory.
`{
    "LISTENIP": "0.0.0.0",
    "PORT": 3218,
    "GIT": false,
    "BASEPATH": "/etc/homeassistant/",
    "ENFORCE_BASEPATH": true,
    "SSL_CERTIFICATE": null,
    "SSL_KEY": null,
    "IGNORE_SSL": false,
    "HASS_API": null,
    "HASS_WS_API": null,
    "HASS_API_PASSWORD": null,
    "USERNAME": null,
    "PASSWORD": null,
    "ALLOWED_NETWORKS": ["192.168.88.0/24","127.0.0.1"],
    "ALLOWED_DOMAINS": [],
    "BANNED_IPS": [],
    "BANLIMIT": 0,
    "IGNORE_PATTERN": ["__pycache__"],
    "DIRSFIRST": true,
    "HIDEHIDDEN": true,
    "SESAME": null,
    "SESAME_TOTP_SECRET": null,
    "VERIFY_HOSTNAME": null,
    "ENV_PREFIX": "HC_",
    "NOTIFY_SERVICE": "persistent_notification.create"
}`
![Снимок экрана в 2021-09-17 13-42-51](https://user-images.githubusercontent.com/883768/133754898-726e7051-066f-4b5d-a1f8-3643f10ab69d.png)
.

**Comment by danielperna84** [comment]
@ravikwow 

Have you tried [this](https://github.com/danielperna84/hass-configurator/issues/171#issuecomment-557316123)? Also please remove the trailing `/` in your `BASEPATH` configuration.

**Comment by ravikwow** [comment]
> @ravikwow
> 
> Have you tried [this](https://github.com/danielperna84/hass-configurator/issues/171#issuecomment-557316123)? Also please remove the trailing `/` in your `BASEPATH` configuration.

Thanks, it work.
